lib/Wikia/src/Service/User/Permissions/PermissionsServiceImpl.php

```diff
@@ -0,0 +1,832 @@
+		$this->globalGroups[] = 'reviewer';
+		$this->globalGroups[] = 'poweruser';
+		$this->globalGroups[] = 'translator';
+		$this->globalGroups[] = 'wikifactory';
+		$this->globalGroups[] = 'restricted-login';
+		$this->globalGroups[] = 'council';
+		$this->globalGroups[] = 'authenticated';
+		$this->globalGroups[] = 'wikiastars';
+		$this->globalGroups[] = 'restricted-login';
+		$this->globalGroups[] = 'voldev';
+	}
+
+	private function loadImplicitGroups() {
+		$this->implicitGroups[] = '*';
+		$this->implicitGroups[] = 'user';
+		$this->implicitGroups[] = 'autoconfirmed';
+		$this->implicitGroups[] = 'poweruser';
+	}
+
+	private function loadExplicitGroups() {
+		global $wgGroupPermissions, $wgRevokePermissions;
+		$this->explicitGroups = array_diff(
+			array_merge( array_keys( $wgGroupPermissions ), array_keys( $wgRevokePermissions ) ),
+			$this->implicitGroups
+		);
+	}
+
+	private function loadPermissions() {
+		$this->permissions = array(
+			'apihighlimits',
+			'autoconfirmed',
+			'autopatrol',
+			'bigdelete',
+			'block',
+			'blockemail',
+			'bot',
+			'browsearchive',
+			'createaccount',
+			'createpage',
+			'createtalk',
+			'delete',
+			'deletedhistory',
+			'deletedtext',
+			'deleterevision',
+			'edit',
+			'editinterface',
+			'editmyoptions',
+			'editusercssjs', #deprecated
+			'editusercss',
+			'edituserjs',
+			'hideuser',
+			'import',
+			'importupload',
+			'ipblock-exempt',
+			'markbotedits',
+			'mergehistory',
+			'minoredit',
+			'move',
+			'movefile',
+			'move-rootuserpages',
+			'move-subpages',
+			'nominornewtalk',
+			'noratelimit',
+			'override-export-depth',
+			'patrol',
+			'protect',
+			'proxyunbannable',
+			'purge',
+			'read',
+			'reupload',
+			'reupload-shared',
+			'rollback',
+			'sendemail',
+			'siteadmin',
+			'suppressionlog',
+			'suppressredirect',
+			'suppressrevision',
+			'unblockself',
+			'undelete',
+			'unwatchedpages',
+			'upload',
+			'upload_by_url',
+			'userrights',
+			'userrights-interwiki',
+			'writeapi',
+			'canremovemap',
+			'wikiawidget',
+			'wikifactory',
+			'wikifactorymetrics',
+			'dumpsondemand',
+			'wikifeatures',
+			'MultiFileUploader',
+			'allowedtoblank',
+			'batchmove',
+			'linkstoredirects',
+			'mobilesearches',
+			'soapfailures',
+			'moderatesotd',
+			'hiderevision',
+			'oversight',
+			'abusefilter-modify',
+			'abusefilter-log-detail',
+			'abusefilter-view',
+			'abusefilter-log',
+			'abusefilter-private',
+			'abusefilter-modify-restricted',
+			'abusefilter-revert',
+			'abusefilter-view-private',
+			'abusefilter-hidden-log',
+			'abusefilter-hide-log',
+			'override-antispoof',
+			'checkuser',
+			'checkuser-log',
+			'geocode',
+			'nuke',
+			'refreshspecial',
+			'replacetext',
+			'spamregex',
+			'tboverride', 	// Implies tboverride-account
+			'tboverride-account', 	// For account creation
+			'torunblocked',
+			'abusefilter-bypass',
+			'platinum',
+			'sponsored-achievements',
+			'achievements-exempt',
```

Comment:
IMHO in such case we need a single source of truth (simple directory?) for the permission names so we can assure there will be no clash in the future when someone adds a new permission not knowing that such one already exists. Makes sense?